source/logging.c

```diff
@@ -351,22 +348,27 @@ void aws_register_log_subject_info_list(struct aws_log_subject_info_list *log_su
     s_log_subject_slots[slot_index] = log_subject_list;
 }
 
-static struct aws_log_subject_info s_common_log_subject_infos[] = {
-    DEFINE_LOG_SUBJECT_INFO(
-        AWS_LS_COMMON_GENERAL,
-        "aws-c-common",
-        "Subject for aws-c-common logging that doesn't belong to any particular category"),
-    DEFINE_LOG_SUBJECT_INFO(
-        AWS_LS_COMMON_TASK_SCHEDULER,
-        "task-scheduler",
-        "Subject for task scheduler or task specific logging."),
-};
+void aws_unregister_log_subject_info_list(struct aws_log_subject_info_list *log_subject_list) {
+    /*
+     * We're not so worried about these asserts being removed in an NDEBUG build
+     * - we'll either segfault immediately (for the first two) or for the count
+     * assert, the registration will be ineffective.
+     */
+    AWS_ASSERT(log_subject_list);
```

Comment:
If you expect to segfault immediatly, why not make it a `FATAL_ASSERT` anyway? Its not like this function is on the critical path of a tight loop.